zaas-client/src/main/java/org/zowe/apiml/zaasclient/config/DefaultZaasClientConfiguration.java

```diff
@@ -52,14 +52,29 @@ public ConfigProperties getConfigProperties() {
         ConfigProperties configProperties = new ConfigProperties();
         configProperties.setApimlHost(host);
         configProperties.setApimlPort(port);
-        configProperties.setApimlBaseUrl(baseUrl);
         configProperties.setKeyStorePath(keyStorePath);
         configProperties.setKeyStorePassword(keyStorePassword);
         configProperties.setKeyStoreType(keyStoreType);
         configProperties.setTrustStorePath(trustStorePath);
         configProperties.setTrustStorePassword(trustStorePassword);
         configProperties.setTrustStoreType(trustStoreType);
         configProperties.setNonStrictVerifySslCertificatesOfServices(nonStrictVerifySslCertificatesOfServices);
+
+        String updatedBaseUrl;
+        if (!baseUrl.startsWith("/gateway") && !baseUrl.startsWith("gateway")) {
```

Comment:
I feel like we missed each other here a bit. 
The original idea was that the baseUrl will be set to the new version, if there is no URL set. 

The transformation of the URL if it is provided is probably a good idea as we don't really support change of serviceId in the core services. 
As for how to do it I would parse the path and verified that he gateway is at the 3rd place and if this is the case I would move it to the first place.